### PR TITLE
IMPROVE SUPPLEJACK MANAGER TEST SPECS

### PR DESCRIPTION
**Pull request by Reinkaos**
Acceptance Criteria

- Controller specs are rewritten without the use of instance_double and double.
- Model specs are rewritten without the use of instance_double and double
- Factories are updated to generate sufficient test data
- Areas without specs are identified and drafted into a follow up story

### CODE REVIEW

File: spec/controllers/abstract_jobs_controller_spec.rb [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+    sign_in create(:user)
   end
 
-  describe "GET index" do
-    it "returns active abstract jobs" do
-      expect(AbstractJob).to receive(:search).with(hash_including("status" => "active"))
-      get :index, status: "active", environment: "staging"
+  describe 'GET index' do
+    it 'returns active abstract jobs' do
+      expect(AbstractJob).to receive(:search).with(hash_including('status' => 'active'))
```
Comment (yarhtut) [inline spec/controllers/abstract_jobs_controller_spec.rb]: maybe the `=>` can be changed to `:`

File: spec/controllers/collection_statistics_controller_spec.rb [hunk 2]
```diff
@@ -9,31 +9,25 @@
...
-		  expect(CollectionStatistics).to receive(:find).with(:all, params: { collection_statistics: {day: Date.today.to_s} }) { collection_statistics }
-		  get :show, id: Date.today.to_s, environment: "staging"
-		  expect(assigns(:collection_statistics)).to eq collection_statistics
-		end
-	end
+  describe 'GET index' do
+    it 'should get all of the collection statistics' do
+      expect(CollectionStatistics).to receive(:all) { [collection_statistics] }
+      get :index, environment: 'staging'
+      expect(assigns(:collection_statistics)).to eq "#{Time.zone.today.to_s}" => {:suppressed=>1, :activated=>0, :deleted=>0}
```
Comment (yarhtut) [inline spec/controllers/collection_statistics_controller_spec.rb]: `=>`

File: spec/controllers/parser_versions_controller_spec.rb [hunk 3]
```diff
@@ -44,45 +49,43 @@
...
     end
   end
 
-  describe "PUT update" do
-    it "updates the version" do
-      expect(version).to receive(:update_attributes).with({ 'tags' => ['staging']})
-      put :update, id: 1, parser_id: 1, version: {tags: ["staging"]}
+  describe 'PUT update' do
+    it 'updates the version' do
+      expect(version).to receive(:update_attributes).with('tags' => ['staging'])
```
Comment (yarhtut) [inline spec/controllers/parser_versions_controller_spec.rb]: `=>`

File: spec/controllers/parsers_controller_spec.rb [hunk 4]
```diff
@@ -26,70 +33,66 @@
...
     before do
       allow(Parser).to receive(:new) { parser }
-      allow(parser).to receive(:save) { true }
     end
 
-    it "initializes a new parser" do
-      expect(Parser).to receive(:new).with({"name" => "Tepapa"}) { parser }
-      post :create, parser: {name: "Tepapa"}
+    it 'initializes a new parser' do
+      expect(Parser).to receive(:new).with({'name' => 'Tepapa'}) { parser }
```
Comment (yarhtut) [inline spec/controllers/parsers_controller_spec.rb]: `=>`

File: spec/controllers/parsers_controller_spec.rb [hunk 5]
```diff
@@ -105,35 +108,35 @@
...
-      put :update, id: "1234", parser: { name: '' }
+      expect(Parser).to receive(:find).with('1234') { parser }
+      put :update, id: '1234', parser: { name: '' }
       expect(assigns(:parser)).to eq parser
     end
 
     it "updates the parser attributes" do
-      expect(parser).to receive("attributes=").with({"name" => "Tepapa"})
-      put :update, id: "1234", parser: {name: "Tepapa"}
+      expect(parser).to receive('attributes=').with({'name' => 'Tepapa'})
```
Comment (yarhtut) [inline spec/controllers/parsers_controller_spec.rb]: `=>`

File: spec/controllers/abstract_jobs_controller_spec.rb (hunk 1 above)
Comment (Reinkaos) [reply spec/controllers/abstract_jobs_controller_spec.rb]: Oh I'm pretty sure you can't when the keys are strings.

**Comment by motizuki** [comment]
@Reinkaos good work 💯

File: spec/controllers/abstract_jobs_controller_spec.rb (hunk 1 above)
Comment (yarhtut) [reply spec/controllers/abstract_jobs_controller_spec.rb]: Thanks bro. good to know

**Pull request merged**